Gems/Profiler/Code/Source/CpuProfilerImpl.cpp

```diff
@@ -329,8 +329,21 @@ namespace Profiler
         {
             return;
         }
-        // Add an entry to the cached region
-        m_cachedTimeRegions.push_back(timeRegionCached);
+        // Add an entry to the cached region. Discard excess data in case there is too much to handle.
+        if (m_cachedTimeRegions.size() < TimeRegionStackSize)
+        {
+            m_cachedTimeRegions.push_back(timeRegionCached);
+        }
+        // Warn only once per thread if the cached data limit has been reached.
+        else if (!m_cachedDataLimitReached)
+        {
+            AZ_Warning(
+                "Profiler", false,
+                "Limit for profiling data has been reached by thread %i. Excess data will be discarded. Considering moving or reducing "
+                "profiler markers to prevent data loss.",
+                m_executingThreadId);
+            m_cachedDataLimitReached = true;
```

Comment:
I imagine this warning could be missed by the user while profiling. Can we update the profiler UI to make it very clear to the user that data is missing?

Reply:
Great idea, let me see what I can do.

Reply:
I'd like more time to add a mechanism for displaying warnings/errors in the profiler UI. I'll be tackling that in the future.